plugins/system/userlogs/userlogs.php

```diff
@@ -129,474 +171,538 @@ public function onContentAfterSave($context, $article, $isNew)
+				}
+				break;
+		}
+	}
+
+	/**
+  * Adds additional fields to the user editing form for logs e-mail notifications
+  *
+  * @param   JForm  $form  The form to be altered.
+  * @param   mixed  $data  The associated data for the form.
+  *
+  * @return  boolean
+  *
+  * @since   1.6
+  */
+	public function onContentPrepareForm($form, $data)
+	{
+		$lang = JFactory::getLanguage();
+		$lang->load('plg_system_userlogs', JPATH_ADMINISTRATOR);
+
+		if (!$form instanceof JForm)
+		{
+			$this->subject->setError('JERROR_NOT_A_FORM');
+
+			return false;
+		}
+
+		if (!in_array($form->getName(), array('com_users.profile', 'com_users.registration','com_users.user','com_admin.profile')))
+		{
+			return true;
+		}
+
+		if ($form->getName() == 'com_admin.profile'
+			|| $form->getName() == 'com_users.profile')
+		{
+			JForm::addFormPath(dirname(__FILE__) . '/profiles');
+			$form->loadFile('profile', false);
+
+			if (!JFactory::getUser()->authorise('core.viewlogs'))
+			{
+				$form->removeField('newsletter_option');
+				$form->removeField('newsletter_extensions');
+			}
+		}
+	}
+
+	/**
+	 * Method called after event log is stored to database
+	 *
+	 * @param   array  $values	The data logged to the database
+	 *
+	 * @return  boolean
+	 */
+	public function onUserLogsAfterMessageLog($message, $date, $context, $user_name, $ip_address)
+	{
+		$dispatcher = JEventDispatcher::getInstance();
+		$query = $this->db->getQuery(true);
+		$query->select('a.email, a.params');
+		$query->from($this->db->quoteName('#__users', 'a'));
+		$query->where($this->db->quoteName('params') . ' LIKE ' . $this->db->quote('%"logs_notification_option":"1"%'));
+		$this->db->setQuery($query);
+		$this->db->execute();
+
+		$users = $this->db->loadObjectList();
+
+		if (empty($users))
+		{
+			return;
+		}
+
+		$recipients = array();
+
+		foreach ($users as $user)
+		{
+			$extensions = json_decode($user->params, true)['logs_notification_extensions'];
+
+			if (in_array(strtok($context, '.'), $extensions))
+			{
+				$recipients[] = $user->email;
+			}
+		}
+
+		if (empty($recipients))
+		{
+			return;
+		}
+
+		$dispatcher->trigger('onLogMessagePrepare', array (&$message, $context));
+		$body = '<h1>'
+			. JText::_('PLG_SYSTEM_USERLOG_EMAIL_SUBJECT') .
+			'</h1><h2>'
+			. JText::_('PLG_SYSTEM_USERLOG_EMAIL_DESC') .
+			'</h2><table><thead>
+					<th>' . JText::_('COM_USERLOGS_MESSAGE') . '</th>
+					<th>' . JText::_('COM_USERLOGS_DATE') . '</th>
+					<th>' . JText::_('COM_USERLOGS_EXTENSION') . '</th>
+					<th>' . JText::_('COM_USERLOGS_USER') . '</th>
+					<th>' . JText::_('COM_USERLOGS_IP_ADDRESS') . '</th>
+				</thead><tbody><tr>
+						<td>' . $message . '</td>
+						<td>' . $log_date . '</td>
+						<td>' . UserlogsHelper::translateExtensionName(strtoupper(strtok($extension), '.')) . '</td>
+						<td>' . $user_name . '</td>
+						<td>' . JText::_($ip_address) . '</td>
+			</tr></tbody></table>';
```

Comment:
That inline html is really.. And you offer no way to override - that needs some improvements.

Maybe use JLayout here!